tracing/src/main/java/io/airlift/tracing/OpenTelemetryModule.java

```diff
@@ -69,12 +74,13 @@ public OpenTelemetry createOpenTelemetry(NodeInfo nodeInfo, OpenTelemetryConfig
 
         Resource resource = Resource.getDefault().merge(Resource.create(attributes.build()));
 
-        SpanExporter spanExporter = OtlpGrpcSpanExporter.builder()
-                .setEndpoint(config.getEndpoint())
-                .build();
-
         SdkTracerProvider tracerProvider = SdkTracerProvider.builder()
-                .addSpanProcessor(BatchSpanProcessor.builder(spanExporter).build())
+                .addSpanProcessor(spanProcessor.orElseGet(() -> {
```

Comment:
If there is a custom `spanProcessor`, the exporter endpoint configuration property is silently ignored.

Reply:
@findepi I've consulted my change with David and he raised similiar concern for my initial attempt

Reply:
You're right. We shouldn't bind the endpoint config when `spanProcessor` is provided. I can address this once we have clarity which way we want to go.